### PR TITLE
TCP-67: Add forms for creating and updating posts

### PR DESCRIPTION
**Pull request by daviddzhang**
Keeping test files in here for now since I think I'll need them for a future task

https://user-images.githubusercontent.com/13986603/148834291-fcdcdb9a-7aaa-45b7-96f9-2ae74511ce1b.mov

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+
+    if @post.save
+      redirect_to @post
+    else
```
Comment (mshenton75) [inline app/controllers/posts_controller.rb:19]: We will usually just roll with the pattern of `Post.create!(...)`, but this works

File: app/views/posts/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+%h1 Posts
+%ul
+  - @posts.each do |post|
+    %li
+      %h4= post.title
+      = link_to "Edit", edit_post_path(post.id)
+      %p= post.text
```
Comment (mshenton75) [inline app/views/posts/index.html.haml:7]: Nice 👏

File: spec/controllers/post_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "GET #new" do
+    it "returns http success" do
+      get :new
+      expect(response).to have_http_status(:success)
+    end
+  end
```
Comment (mshenton75) [inline spec/controllers/post_controller_spec.rb:17]: Shouldn't this be a `POST` route?

File: spec/controllers/post_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  describe "GET #update" do
+    it "returns http success" do
+      get :update
+      expect(response).to have_http_status(:success)
+    end
+  end
```
Comment (mshenton75) [inline spec/controllers/post_controller_spec.rb:31]: Same with this, I'd expect it be a `PUT` route.  Do the tests pass?

File: spec/factories/posts.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :post do
+    title { "MyString" }
```
Comment (mshenton75) [inline spec/factories/posts.rb:3]: We use a library `Faker` in production for false data so you could do `title { Faker::Lorem.word }` . . . I'm not sure if it's installed in Launchpad but you could try it, and even add it to the project as an improvement if it's not installed, if you want 😎

**Review by mshenton75 — APPROVED** [review]
LGTM, though I would still advise against creating all this scaffolding through generators, because it does create a lot of bloat.  For example, we don't really use anything like `spec/views` tests in the `app`. I think all in all, it's better to just create each file manually when you need it.  

Nice work on this!

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (daviddzhang) [reply app/controllers/posts_controller.rb:19]: Oh, I forgot that was a thing. Will change, thanks!

File: spec/controllers/post_controller_spec.rb (hunk 3 above)
Comment (daviddzhang) [reply spec/controllers/post_controller_spec.rb:17]: These generated tests might be out sync with the actual code after making some manual changes. But I think this one is okay as a GET. You can visit localhost:3000/posts/new to get to the page with the new post form. I believe sending a POST to /posts will create a post

File: spec/controllers/post_controller_spec.rb (hunk 4 above)
Comment (daviddzhang) [reply spec/controllers/post_controller_spec.rb:31]: No they do not 😅 I would normally fix it in this PR, but the next task is to write tests for post creation, so I think it makes more sense just to do it in a separate PR so I can have one PR per task.

File: spec/factories/posts.rb (hunk 5 above)
Comment (daviddzhang) [reply spec/factories/posts.rb:3]: Good to know! Looks like it's already in the Gemfile, so I'll play around with it in the next task when I write the tests

**Pull request merged**